src/java.base/share/classes/jdk/internal/lang/stable/StableFieldUpdater.java

```diff
@@ -0,0 +1,365 @@
+     * @param underlying      function to filter return values from
+     * @param zeroReplacement to replace any zero values returned by the {@code underlying}
+     *                        method handle.
+     */
+    public static MethodHandle replaceLongZero(MethodHandle underlying, long zeroReplacement) {
+
+        final class Holder {
+            private static final MethodHandle RETURN_FILTER =
+                    MhUtil.findStatic(LOCAL_LOOKUP, "replaceZero", MethodType.methodType(long.class, long.class, long.class));
+        }
+
+        check(underlying, long.class);
+        return MethodHandles.filterReturnValue(underlying,
+                MethodHandles.insertArguments(Holder.RETURN_FILTER, 1, zeroReplacement));
+    }
+
+    /**
+     * {@return a method handle that will replace any null value returned by the provided
+     *          {@code underlying} handle with the provided {@code nullReplacement}}
+     * @param underlying      function to filter return values from
+     * @param nullReplacement to replace any zero values returned by the {@code underlying}
+     *                        method handle.
+     */
+    public static MethodHandle replaceReferenceNull(MethodHandle underlying, Object nullReplacement) {
+
+        final class Holder {
+            private static final MethodHandle RETURN_FILTER =
+                    MhUtil.findStatic(LOCAL_LOOKUP, "replaceNull", MethodType.methodType(Object.class, Object.class, Object.class));
+        }
+        check(underlying, Object.class);
+        return MethodHandles.filterReturnValue(underlying,
+                MethodHandles.insertArguments(Holder.RETURN_FILTER, 1, nullReplacement));
+    }
+
+    // Used reflectively
+    @ForceInline
+    private static int replaceZero(int value, int zeroReplacement) {
+        return value == 0 ? zeroReplacement : value;
+    }
+
+    // Used reflectively
+    @ForceInline
+    private static long replaceZero(long value, long zeroReplacement) {
+        return value == 0 ? zeroReplacement : value;
+    }
+
+    // Used reflectively
+    // Cannot reuse `Objects::requireNonNullElse` as it prohibits `null` replacements
+    @ForceInline
+    private static Object replaceNull(Object value, Object nullReplacement) {
+        return value == null ? nullReplacement : value;
+    }
+
+    public static CallSite lazyAtMostOnce(MethodHandles.Lookup lookup,
+                                          String unused,
+                                          VarHandle accessor,
+                                          MethodHandle underlying) {
+        Objects.requireNonNull(accessor);
+        Objects.requireNonNull(underlying);
+        var handle = MhUtil.findStatic(LOCAL_LOOKUP,
+                "atMostOnce", MethodType.methodType(MethodHandle.class, VarHandle.class, MethodHandle.class));
+        return new ConstantCallSite(MethodHandles.insertArguments(handle, 0, accessor, underlying));
```

Comment:
This makes no sense as a bootstrap for a dynamic call site, as that’ll simply be equivalent to a call to `atMostOnce`.

I presume thou meant something like the following, which would actually create the stable field updater for use at the call site:
```suggestion
        return new ConstantCallSite(atMostOnce(accessor, underlying).asType(type));
```

or by using the `createTargetHook` of `ConstantCallSite`:
```suggestion
        var handle = MhUtil.findStatic(LOCAL_LOOKUP,
                "atMostOnce", MethodType.methodType(MethodHandle.class, VarHandle.class, MethodHandle.class));
        return new ConstantCallSite(type, MethodHandles.dropArguments(
                MethodHandles.insertArguments(handle, 0, accessor, underlying),
                0, ConstantCallSite.class));
```